test/hotspot/jtreg/compiler/vectorapi/reshape/tests/TestVectorDoubleExpandShrink.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi.reshape.tests;
+
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.Run;
+import compiler.lib.ir_framework.Test;
+
+import static compiler.vectorapi.reshape.utils.VectorReshapeHelper.*;
+
+/**
+ * As spot in 8259353. We need to do a shrink and an expand together to not accidentally
+ * zero out elements in the physical registers that may not be zero in general cases.
+ * In some methods, 2 consecutive ReinterpretNodes may be optimized out.
+ */
+public class TestVectorDoubleExpandShrink {
+    @Test
+    @IR(counts = {REINTERPRET_NODE, "0"})
```

Comment:
You could use the equivalent attribute `failOn = REINTERPRET_NODE` for a zero count instead.